agent/src/beerocks/monitor/monitor_thread.h

```diff
@@ -59,6 +59,23 @@ class monitor_thread : public beerocks::socket_thread {
     void stop_monitor_thread();
     bool hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event_ptr);
 
+    /**
+     * @brief Executes when channel utilization measurement period has elapsed.
+     *
+     * This method is periodically invoked if AP Metrics Channel Utilization Reporting Threshold
+     * has been set to a non-zero value in last received Metric Reporting Policy TLV.
+     * Measurement period is set to an implementation-specific value.
+     * On invocation, method shall measure current channel utilization on the radio. If difference
+     * with respect to the previous measurement has crossed the reporting threshold, it shall send
+     * an AP Metrics Response message to the controller.
+     */
+    void on_channel_utilization_measurement_period_elapsed();
+
+    /**
+     * @brief Creates AP Metrics Response message
+     */
+    bool create_ap_metrics_response();
```

Comment:
I put this function in monitor_stats. It would read its information from the database anyway, so putting it in monitor_stats is fine. So on_channel_utilization_measurement_period_elapsed would update the database before calling create_ap_metrics_response(). And I think ti's pretty easy: ap_metrics_channel_utilization_reporting_value and _time should move to sRadioStats. Note that that object already has channel_load field which AFAIU are more or less equivalent (but apparently never got implemented because channel_load_tot_curr is never set and all the rest depends on it).

That said, I'm perfectly fine with moving create_ap_metrics_response to monitor_thread. Like you, I don't see a clear separation of concerns between the monitor classes (except for the db, of course).

Note that the response needs a list of bssids that have to be included in the response.

Reply:
My create_ap_metrics_response() method is just a hint of how/where it could be done. 
This is something left to be done when merging your branch and mine